osmoutils/store_helper_test.go

```diff
@@ -186,3 +189,220 @@ func (s *TestSuite) TestNoStopFn_AlwaysFalse() {
 	s.Require().False(osmoutils.NoStopFn([]byte(keyA)))
 	s.Require().False(osmoutils.NoStopFn([]byte(keyB)))
 }
+
+// TestMustGet tests that MustGet retrieves the correct
+// values from the store and panics if an error is encountered.
+func (s *TestSuite) TestMustGet() {
+
+	tests := map[string]struct {
+		// keyys and values to preset
+		preSetKeyValues map[string]proto.Message
+
+		// keys and values to attempt to get and validate
+		getKeyValues map[string]proto.Message
+
+		actualResultProto proto.Message
+
+		expectPanic bool
+	}{
+		"basic valid test": {
+			preSetKeyValues: map[string]proto.Message{
+				keyA: &sdk.DecProto{Dec: sdk.OneDec()},
+				keyB: &sdk.DecProto{Dec: sdk.OneDec().Add(sdk.OneDec())},
+				keyC: &sdk.DecProto{Dec: sdk.OneDec().Add(sdk.OneDec())},
+			},
+
+			getKeyValues: map[string]proto.Message{
+				keyA: &sdk.DecProto{Dec: sdk.OneDec()},
+				keyB: &sdk.DecProto{Dec: sdk.OneDec().Add(sdk.OneDec())},
+				keyC: &sdk.DecProto{Dec: sdk.OneDec().Add(sdk.OneDec())},
+			},
+
+			actualResultProto: &sdk.DecProto{},
+		},
+		"attempt to get non-existent key - panic": {
+			preSetKeyValues: map[string]proto.Message{
+				keyA: &sdk.DecProto{Dec: sdk.OneDec()},
+				keyC: &sdk.DecProto{Dec: sdk.OneDec().Add(sdk.OneDec())},
+			},
+
+			getKeyValues: map[string]proto.Message{
+				keyB: &sdk.DecProto{Dec: sdk.OneDec().Add(sdk.OneDec())},
+			},
+
+			actualResultProto: &sdk.DecProto{},
+
+			expectPanic: true,
+		},
+		"invalid proto Dec vs TwapRecord- error": {
+			preSetKeyValues: map[string]proto.Message{
+				keyA: &sdk.DecProto{Dec: sdk.OneDec()},
+			},
+
+			getKeyValues: map[string]proto.Message{
+				keyA: &sdk.DecProto{Dec: sdk.OneDec()},
+			},
+
+			actualResultProto: &twaptypes.TwapRecord{},
+
+			expectPanic: true,
+		},
+	}
+
+	for name, tc := range tests {
+		s.Run(name, func() {
+			s.SetupStoreWithBasePrefix()
+
+			// Setup
+			for key, value := range tc.preSetKeyValues {
+				osmoutils.MustSet(s.store, []byte(key), value)
+			}
+
+			osmoassert.ConditionalPanic(s.T(), tc.expectPanic, func() {
+				for key, expectedValue := range tc.getKeyValues {
+					// System under test.
+					osmoutils.MustGet(s.store, []byte(key), tc.actualResultProto)
+					// Assertions.
+					s.Require().Equal(expectedValue.String(), tc.actualResultProto.String())
```

Comment:
All the logic here makes sense, just had a design question. 

Normally within the table driven tests we have done, we specify the expected value in the table and then within the test logic itself we get the actual value and compare the two. It seems here that those events were flipped. Was this intentional / does it not really matter? Or should we stick to the same way for all table driven tests?

Reply:
The intention here was to test multiple gets in sequence, that's why these are surrounded by a loop and the expected is map containing multiple expected values.

I renamed `tc.GetKeyValues` to `tc.expectedGetKeyValues` to make it clearer. Let me know what you think

Reply:
yeah that is clearer now, thanks!